cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java

```diff
@@ -66,11 +67,7 @@ public KafkaMetadataStateManager(
         this.kraftAnno = kraftAnnotation(kafkaCr);
         KafkaMetadataState metadataStateFromKafkaCr = kafkaCr.getStatus() != null ? kafkaCr.getStatus().getKafkaMetadataState() : null;
         // missing metadata state means reconciling an already existing Kafka resource with newer operator supporting metadata state or first reconcile
-        if (metadataStateFromKafkaCr == null) {
-            this.metadataState = isKRaftAnnoEnabled() ? KafkaMetadataState.KRaft : KafkaMetadataState.ZooKeeper;
-        } else {
-            this.metadataState = metadataStateFromKafkaCr;
-        }
+        this.metadataState = Objects.requireNonNullElseGet(metadataStateFromKafkaCr, () -> isKRaftAnnoEnabled() ? KafkaMetadataState.KRaft : KafkaMetadataState.ZooKeeper);
```

Comment:
Why this change? Wasn't the original code easier to read while doing the same?

Reply:
I see this code shorter for a simple thing and understandable at the same time like the ternary operator.

Reply:
@strimzi/maintainers any more opinions on this?

Reply:
Well, the old one was more readable because we are used to this I think :) the new one is just a fancy one-liner...I don't have a strong opinion about what is better...but for me, the readability is better in the old one but because I am more used to that form...

Reply:
I agree that we are not much (I found one occurrence in STs) `requireNonNullElseGet` in the project and the previous format was less confusing FMPOV (as mentioned by @see-quick , just because we are not using it regularly).

Also, if you don't want to have the `else` block there, you can do
```
this.metadataState = metadataStateFromKafkaCr;

if (metadataStateFromKafkaCr == null) {
    this.metadataState = isKRaftAnnoEnabled() ? KafkaMetadataState.KRaft : KafkaMetadataState.ZooKeeper;
}
```
which will be even more confusing 😄 but it's an option.

Reply:
ok, I reverted back to the previous code.